osrm/OSRM_nodejs_bindings.md

```diff
@@ -0,0 +1,265 @@
+
+For a comprehensive feature list and a more general overview of FOSS routing engines, read our [article](https://gis-ops.com/open-source-routing-engines-and-algorithms-an-overview/).
+
+While we are interested in the matrix (table) capabilities, you also have the option to use its other services, too.
+
+- [routing](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/nodejs/api.md#route)
+- [matrix](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/nodejs/api.md#table)
+- [nearest points](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/nodejs/api.md#nearest)
+- [map tiles](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/nodejs/api.md#tile)
+- [map matching](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/nodejs/api.md#match)
+- [traveling salesman problem](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/nodejs/api.md#trip)
+
+ > **Disclaimer**  
+ > Validity only confirmed for **Ubuntu 20.04** / **Mac OSX 10.15.6** and **[OSRM 5.26.0](https://www.npmjs.com/package/osrm)**.
+
+### Recommended
+
+-  A basic understanding of Javascript, NodeJS and npm
+-  The awareness that a red printed terminal line doesn't necessarily mean, it's an error ;)...
+
+This tutorial will guide you through the process of how to configure and run the NodeJS bindings for OSRM. 
+
+So let's go, `open your terminal!`
+
+## 1. Preparations
+
+### Test your NodeJS & npm installation
+
+Test whether your environment already has NodeJS installed.
+
+```bash
+node -v
+```
+
+Similarly test if your environment has Node's package manager npm installed.
+
+```bash
+npm -v
+```
+
+If either output is similar to this:
+
+```bash
+command not found: ...
+```
+
+You will have to **install nodejs and/or npm first**. 
+To do this have a look at the [How To Install Node.js on Ubuntu 20.04](https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-20-04) or if you are running OSX head over to [Install Node.js and npm using Homebrew on OSX and macOS](https://changelog.com/posts/install-node-js-with-homebrew-on-os-x).
+
+
+### Create your Working Directory and get NPM ready
+
+First create a working directory:
+```bash
+mkdir ~/gisops_osrm_nodejs && cd $_
+```   
+
+In order to install the required dependencies for this tutorial you will need to copy and paste the following snippet into your [package.json](https://docs.npmjs.com/cli/v8/configuring-npm/package-json) file. 
+
+```bash
+{
+  "name": "osrm node bindings gis-ops tutorial",
+  "version": "0.1",
+  "dependencies": {
+    "async": "^3.2.2",
+    "osrm": "^5.26.0"
+  }
+}
+```
+
+We are making use of 2 libraries [osrm](https://www.npmjs.com/package/osrm) and [async](https://www.npmjs.com/package/async) which will allow us to use the OSRM nodejs bindings in a multithreaded fashion.
+With the following command you instruct npm to install these dependencies which will read the `package.json` file sitting in the same folder.
```

Comment:
I'm missing a few words about how the npm package contains all binaries & config files we need to run OSRM. for the not-so-technical reader it might be mind-blowing that all this stuff is in a npm package